core/src/main/java/com/netflix/conductor/service/ExecutionLockService.java

```diff
@@ -0,0 +1,85 @@
+package com.netflix.conductor.service;
+
+import com.netflix.conductor.common.run.Workflow;
+import com.netflix.conductor.core.config.Configuration;
+import com.netflix.conductor.core.utils.Lock;
+import com.netflix.conductor.metrics.Monitors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import java.util.concurrent.TimeUnit;
+
+public class ExecutionLockService {
+    public static final String LOCK_NAMESPACE = "executionlock";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExecutionLockService.class);
+    private final Configuration config;
+    private final Provider<Lock> lockProvider;
+    private static long LOCK_TIME_TO_TRY;
+    private static long LOCK_LEASE_TIME;
+
+    @Inject
+    public ExecutionLockService(Configuration config, Provider<Lock> lockProvider) {
+        this.config = config;
+        this.lockProvider = lockProvider;
+        LOCK_LEASE_TIME = config.getLongProperty("locking.leaseTimeInMilliSeconds", 60000);
```

Comment:
Consider namespacing and casing the properties, for example - 
`workflow.locking.lease.time.ms`